src/theory/fp/theory_fp_rewriter.cpp

```diff
@@ -788,6 +789,95 @@ namespace constantFold {
   }
 
 
+
+  RewriteResponse componentFlag (TNode node, bool) {
+    Kind k = node.getKind();
+
+    Assert((k == kind::FLOATINGPOINT_COMPONENT_NAN ) ||
+	   (k == kind::FLOATINGPOINT_COMPONENT_INF ) ||
+	   (k == kind::FLOATINGPOINT_COMPONENT_ZERO ) ||
+	   (k == kind::FLOATINGPOINT_COMPONENT_SIGN ));
+
+    FloatingPoint arg0(node[0].getConst<FloatingPoint>());
+
+    bool result;
+    switch (k) {
+    case kind::FLOATINGPOINT_COMPONENT_NAN :
+      result = arg0.getLiteral().nan;
+      break;
+    case kind::FLOATINGPOINT_COMPONENT_INF :
+      result = arg0.getLiteral().inf;
+      break;
+    case kind::FLOATINGPOINT_COMPONENT_ZERO :
+      result = arg0.getLiteral().zero;
+      break;
+    case kind::FLOATINGPOINT_COMPONENT_SIGN :
+      result = arg0.getLiteral().sign;
+      break;
+    default :
+      Unreachable("Unknown kind used in constantoldComponentFlag");
```

Comment:
Is this supposed to be "constantoldComponentFlag" or is it a typo (and it should read "constantFoldComponentFlag") or should this be "componentFlag"?

Reply:
Fixed.